File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/configuration/SchedulingConfiguration.java
package com.jmsoftware.maf.springcloudstarter.configuration;

import com.jmsoftware.maf.springcloudstarter.property.MafProjectProperties;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.autoconfigure.task.TaskSchedulingAutoConfiguration;
import org.springframework.boot.task.TaskSchedulerCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.scheduling.annotation.EnableScheduling;

import static cn.hutool.core.text.CharSequenceUtil.format;

/**
 * <h1>SchedulingConfiguration</h1>
 * <p>
 * Change description here.
 *
 * @author Johnny Miller (鍾俊), email: johnnysviva@outlook.com, 10/2/21 2:30 PM
 * @see TaskSchedulingAutoConfiguration
 **/
@Slf4j
@EnableScheduling
@RequiredArgsConstructor
public class SchedulingConfiguration {
    public static final int TASK_SCHEDULER_POOL_SIZE = 2;
    private final MafProjectProperties mafProjectProperties;

    @Bean
    public TaskSchedulerCustomizer taskSchedulerCustomizer() {
        return taskScheduler -> {
            taskScheduler.setPoolSize(TASK_SCHEDULER_POOL_SIZE);
            taskScheduler.setThreadNamePrefix(
                    format("{}-task-scheduler-", this.mafProjectProperties.getProjectArtifactId())
            );
            taskScheduler.setAwaitTerminationSeconds(10);
            log.warn("Enhanced taskScheduler. poolSize = {}", taskScheduler.getPoolSize());
        };
    }
}

File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/configuration/SchedulingConfiguration.kt
package com.jmsoftware.maf.springcloudstarter.configuration

import com.jmsoftware.maf.common.util.logger
import com.jmsoftware.maf.springcloudstarter.property.MafProjectProperties
import org.springframework.boot.autoconfigure.task.TaskSchedulingAutoConfiguration
import org.springframework.boot.task.TaskSchedulerCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler

/**
 * # SchedulingConfiguration
 *
 * Change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/14/22 10:58 AM
 * @see TaskSchedulingAutoConfiguration
 */
@EnableScheduling
class SchedulingConfiguration(
    private val mafProjectProperties: MafProjectProperties
) {
    companion object {
        const val TASK_SCHEDULER_POOL_SIZE = 2
        private val log = logger()
    }

    @Bean
    fun taskSchedulerCustomizer(): TaskSchedulerCustomizer {
        return TaskSchedulerCustomizer { taskScheduler: ThreadPoolTaskScheduler ->
            taskScheduler.poolSize = TASK_SCHEDULER_POOL_SIZE
            taskScheduler.setThreadNamePrefix("${mafProjectProperties.projectArtifactId}-task-scheduler-")
            taskScheduler.setAwaitTerminationSeconds(10)
            log.warn("Enhanced taskScheduler. poolSize = {}", taskScheduler.poolSize)
        }
    }
}
